docs/source/asr/examples/esperanto_asr/scripts/convert_hf_dataset_to_nemo_v2.py

```diff
@@ -0,0 +1,411 @@
+    audio_filepath = audio_filepath + '.wav'
+
+    # Remove previous run file
+    if os.path.exists(audio_filepath):
+        os.remove(audio_filepath)
+    return audio_filepath
+
+
+def build_map_dataset_to_nemo_func(cfg: HFDatasetConvertionConfig, basedir):
+    """
+    Helper method to run in batch mode over a mapped Dataset.
+
+    Creates a function that can be passed to Dataset.map() containing the config and basedir.
+    Useful to map a HF dataset to NeMo compatible format in an efficient way for offline processing.
+
+    Returns:
+        A function pointer which can be used for Dataset.map()
+    """
+
+    def map_dataset_to_nemo(batch):
+        # Write audio file to correct path
+        if cfg.streaming:
+            batch['audio_filepath'] = batch['audio']['path'].split("::")[0].replace("zip://", "")
+        else:
+            segments = infer_dataset_segments(batch)
+            audio_filepath = os.path.join(*segments)
+            batch['audio_filepath'] = audio_filepath
+
+        batch['audio_filepath'] = os.path.abspath(os.path.join(basedir, batch['audio_filepath']))
+        audio_filepath = batch['audio_filepath']
+        audio_filepath = prepare_audio_filepath(audio_filepath)
+        batch['audio_filepath'] = audio_filepath  # update filepath with prepared path
+
+        soundfile.write(audio_filepath, batch['audio']['array'], samplerate=cfg.sampling_rate, format='wav')
+
+        batch['duration'] = librosa.get_duration(y=batch['audio']['array'], sr=batch['audio']['sampling_rate'])
+        return batch
+
+    return map_dataset_to_nemo
+
+
+def convert_offline_dataset_to_nemo(
+    dataset: Dataset, cfg: HFDatasetConvertionConfig, basedir: str, manifest_filepath: str
+):
+    """
+    Converts a HF dataset to a audio-preprocessed Nemo dataset in Offline mode.
+    Also writes out a nemo compatible manifest file.
+
+    Args:
+        dataset: Iterable HF Dataset.
+        cfg: HFDatasetConvertionConfig.
+        basedir: Base output directory.
+        manifest_filepath: Filepath of manifest.
+    """
+    num_proc = cfg.num_proc
+    if num_proc < 0:
+        num_proc = max(1, os.cpu_count() // 2)
+
+    dataset = dataset.map(build_map_dataset_to_nemo_func(cfg, basedir), num_proc=num_proc)
+    ds_iter = iter(dataset)
+
+    with open(manifest_filepath, 'w') as manifest_f:
+        for idx, sample in enumerate(
+            tqdm.tqdm(
+                ds_iter, desc=f'Processing {cfg.path} (split : {cfg.split}):', total=len(dataset), unit=' samples'
+            )
+        ):
+            # remove large components from sample
+            manifest_f.write(
+                json.dumps(
+                    {
+                        "audio_filepath": sample["audio_filepath"],
+                        "duration": sample["duration"],
+                        "text": sample["sentence"],
+                    },
+                    ensure_ascii=False,
+                )
```

Comment:
Hey @titu1994 are we ok with enforcing this manifest path entry in the main script? I.e just these fields get added to the manifest. This seems fine as all other manifests for ASR rely on this format but wanted to check if there are any edge cases.

This would allow us to merge the custom logic here into the main script (and remove this separate script).